src/taipy/core/submission/submission.py

```diff
@@ -0,0 +1,180 @@
+    __SEPARATOR = "_"
+
+    def __init__(
+        self,
+        entity_id: str,
+        id: Optional[str] = None,
+        jobs: Optional[Union[List[Job], List[JobId]]] = None,
+        creation_date: Optional[datetime] = None,
+        submission_status: Optional[SubmissionStatus] = None,
+        version: Optional[str] = None,
+    ):
+        self._entity_id = entity_id
+        self.id = id or self.__new_id()
+        self._jobs: Union[List[Job], List[JobId], List] = jobs or []
+        self._creation_date = creation_date or datetime.now()
+        self._submission_status = submission_status or SubmissionStatus.SUBMITTED
+        self._version = version or _VersionManagerFactory._build_manager()._get_latest_version()
+
+    @staticmethod
+    def __new_id() -> str:
+        """Generate a unique Submission identifier."""
+        return SubmissionId(Submission.__SEPARATOR.join([Submission._ID_PREFIX, str(uuid.uuid4())]))
+
+    @property
+    def entity_id(self) -> str:
+        return self._entity_id
+
+    @property
+    def creation_date(self):
+        return self._creation_date
+
+    def get_label(self) -> str:
+        """Returns the submission simple label prefixed by its owner label.
+
+        Returns:
+            The label of the submission as a string.
+        """
+        return self._get_label()
+
+    def get_simple_label(self) -> str:
+        """Returns the submission simple label.
+
+        Returns:
+            The simple label of the submission as a string.
+        """
+        return self._get_simple_label()
+
+    @property  # type: ignore
+    @_self_reload(_MANAGER_NAME)
+    def jobs(self) -> List[Job]:
+        jobs = []
+        job_manager = _JobManagerFactory._build_manager()
+
+        for job in self._jobs:
+            jobs.append(job_manager._get(job))
+
+        return jobs
+
+    @jobs.setter  # type: ignore
+    @_self_setter(_MANAGER_NAME)
+    def jobs(self, jobs: Union[List[Job], List[JobId]]):
+        self._jobs = jobs
+
+    def __hash__(self):
+        return hash(self.id)
+
+    def __eq__(self, other):
+        return self.id == other.id
+
+    @property  # type: ignore
+    @_self_reload(_MANAGER_NAME)
+    def submission_status(self):
+        return self._submission_status
+
+    @submission_status.setter  # type: ignore
+    @_self_setter(_MANAGER_NAME)
+    def submission_status(self, submission_status):
+        self._submission_status = submission_status
+
+    def __lt__(self, other):
+        return self.creation_date.timestamp() < other.creation_date.timestamp()
+
+    def __le__(self, other):
+        return self.creation_date.timestamp() == other.creation_date.timestamp() or self < other
+
+    def __gt__(self, other):
+        return self.creation_date.timestamp() > other.creation_date.timestamp()
+
+    def __ge__(self, other):
+        return self.creation_date.timestamp() == other.creation_date.timestamp() or self > other
+
+    def _update_submission_status(self, _: Job):
```

Comment:
⚠️ This method is really hard to read and hard to understand. 
Why not something like this?
```
  if job.is_canceled():
      self.submission_status = SubmissionStatus.CANCELED
  if job.is_blocked():
      self.submission_status = SubmissionStatus.BLOCKED
  if job.is_pending() or job.is_submitted():
      self.submission_status = SubmissionStatus.PENDING
  if job.is_running():
      self.submission_status = SubmissionStatus.RUNNING
  if job.is_completed() or job.is_skipped():
      self.submission_status = SubmissionStatus.COMPLETED
  if job.is_abandoned():
      self.submission_status = SubmissionStatus.UNDEFINED
```

Reply:
I believe that's so we can loop through all of the jobs and prioritize some of the statuses. Like for example, if we have a list of canceled jobs, but 1 of them is a failed job, we would want to prioritize the failing status instead of the cancel one. Hope it make sense! 

@jrobinAV can you help fill in if I missed any detail?

Reply:
I don't know what to say. 😄 
The behavior of your code proposal @gmarabout, is not the same as what we implemented. 
It does not respect what we want to implement.
We have to parse all the jobs before computing the submission_status.

Reply:
> It does not respect what we want to implement.
That's is my point: the intent of the method is not obvious when you read it.

Reply:
@gmarabout gm
 would renaming it to `_update_submission_status_based_on_jobs` help?

Reply:
Definitely not :-) 
The name should reflect the intent of the method, what it is supposed to do.
Not *how* it does it. 
What if we decide to change the logic in the future?
The code should definitely speak by itself about the *how*...
If you think the code is clear enough, then let's go. I don't want to block this PR.

Reply:
alright thanks Gregoire! I'll merge it for now and think how i can improve it later